pkg/systemlogmonitor/log_monitor.go

```diff
@@ -165,7 +200,11 @@ func (l *logMonitor) generateStatus(logs []*logtypes.Log, rule systemlogtypes.Ru
 				break
 			}
 		}
+		if rule.MetricName != "" {
+			l.gauges[rule.MetricName].Set(rule.Reason)
```

Comment:
We should also initialize the problem gauges in `initializeStatus()` (line 217).

Reply:
Sounds good. However, I think we can just initialize the metrics at the definition time.
i.e. When caller calls `NewProblemCounter()` or `NewProblemGauge()`, we automatically assign the first measurement for them (i.e. `0`).

That will result in below UX:
```
$ curl http://localhost:20257/metrics
...
# HELP kernel_deadlock Whether below pattern has been observed in log source "kernel-monitor" or not: "task umount\\\\.aufs:\\\\w+ blocked for more than \\\\w+ seconds\\\\."
# TYPE kernel_deadlock gauge
kernel_deadlock{reason=""} 0
# HELP kernel_oom_killing Number of times below pattern is observed in log source "kernel-monitor": "Kill process \\\\d+ (.+) score \\\\d+ or sacrifice child\\\\nKilled process \\\\d+ (.+) total-vm:\\\\d+kB, anon-rss:\\\\d+kB, file-rss:\\\\d+kB.*"
# TYPE kernel_oom_killing counter
kernel_oom_killing{reason=""} 0
# HELP kernel_oops Number of times below pattern is observed in log source "kernel-monitor": "BUG: unable to handle kernel NULL pointer dereference at .*"
# TYPE kernel_oops counter
kernel_oops{reason=""} 0
# HELP kernel_readonly_filesystem Whether below pattern has been observed in log source "kernel-monitor" or not: "Remounting filesystem read-only"
# TYPE kernel_readonly_filesystem gauge
kernel_readonly_filesystem{reason=""} 0
# HELP kernel_task_hung Number of times below pattern is observed in log source "kernel-monitor": "task \\\\S+:\\\\w+ blocked for more than \\\\w+ seconds\\\\."
# TYPE kernel_task_hung counter
kernel_task_hung{reason=""} 0
# HELP kernel_unregister_netdevice Number of times below pattern is observed in log source "kernel-monitor": "unregister_netdevice: waiting for \\\\w+ to become free. Usage count = \\\\d+"
# TYPE kernel_unregister_netdevice counter
kernel_unregister_netdevice{reason=""} 0
```

Does this looks like what you expected?